### ISSUE TITLE
Code for SA1135 UsingDirectivesMustBeQualified results in unbuildable code for a nested type within a generic type

### ISSUE THREAD

**Issue by davkean** [DotNetAnalyzers/StyleCopAnalyzers#2879]
_From @davkean on February 13, 2019 6:45_

``` C# 
namespace Microsoft.VisualStudio.ProjectSystem.Designers
{
    using System;
    using System.Collections.Generic;
    using System.Collections.Immutable;
    using System.Diagnostics;
    using System.Diagnostics.Contracts;
    using System.IO;
    using System.Linq;
    using System.Threading;
    using Microsoft.VisualStudio.ProjectSystem;
    using Microsoft.VisualStudio.ProjectSystem.Properties;
 
    using NodeLookupTableBuilder = System.Collections.Immutable.ImmutableDictionary<System.IntPtr, System.Tuple<UnattachedProjectTreeNode, System.IntPtr>>.Builder;
}
```

Expected: 

``` C#
    using NodeLookupTableBuilder = System.Collections.Immutable.ImmutableDictionary<System.IntPtr, System.Tuple<Microsoft.VisualStudio.ProjectSystem.Designers.UnattachedProjectTreeNode, System.IntPtr>>.Builder;
```

Actual:

``` C#
    using NodeLookupTableBuilder = System.Collections.Immutable.Builde>;
```

_Copied from original issue: StyleCop/StyleCop#211_

**Comment by davkean** [comment]
Looks like the warning has the wrong name in the message:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	SA1135	Using directive for type 'System.Collections.Immutable.Builde>' should be qualified	Microsoft.VisualStudio.ProjectSystem.Implementation	E:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\Designers\UnattachedProjectTreeNode.cs	20	Active
```

**Comment by davkean** [comment]
_From @SanjayGuntur on February 13, 2019 8:12_

Can you provide more info? 
- Stylecop Version
- R# Version

**Comment by davkean** [comment]
Looks like this actually belongs in https://github.com/DotNetAnalyzers/StyleCopAnalyzers? The NuGet & Roslyn based analyzers are different to StyleCop is that right?

**Comment by davkean** [comment]
_From @SanjayGuntur on February 13, 2019 11:39_

Right, this repo is now referred to as "Stylecop Classic". You should post this issue over at the analyzer repo.
